tests/test_backend.py

```diff
@@ -157,32 +161,119 @@ class TestAggregatorSecondaryServices:
+        backend1,
+        backend2,
+        requests_mock,
+    ):
+        requests_mock.get(backend1 + "/", json=JSON_CAPABILITIES_NO_ENDPOINTS)
+        requests_mock.get(
+            backend2 + "/", json=JSON_CAPABILITIES_WITH_SERVICE_TYPES_SUPPORTED
+        )
+        processing = AggregatorProcessing(
+            backends=multi_backend_connection, catalog=catalog, config=config
+        )
+        implementation = AggregatorSecondaryServices(
+            backends=multi_backend_connection, processing=processing, config=config
+        )
+
+        actual_supported_backends = implementation.get_supporting_backend_ids()
+        assert actual_supported_backends == ["b2"]
+
     def test_service_types_simple(
-        self, multi_backend_connection, config, catalog, backend1, backend2, requests_mock
+        self,
+        multi_backend_connection,
+        config,
+        catalog,
+        backend1,
+        backend2,
+        requests_mock,
     ):
         """Given 2 backends and only 1 backend has a single service type, then the aggregator
             returns that 1 service type's metadata.
         """
+        # Aggregator checks if the backend supports GET /service_types, so we have to mock that up too.
+        requests_mock.get(
+            backend1 + "/", json=JSON_CAPABILITIES_WITH_SERVICE_TYPES_SUPPORTED
+        )
+        requests_mock.get(
+            backend2 + "/", json=JSON_CAPABILITIES_WITH_SERVICE_TYPES_SUPPORTED
+        )
         single_service_type = self.SERVICE_TYPES_ONLT_WMTS
         requests_mock.get(backend1 + "/service_types", json=single_service_type)
         requests_mock.get(backend2 + "/service_types", json={})
+
         processing = AggregatorProcessing(backends=multi_backend_connection, catalog=catalog, config=config)
         implementation = AggregatorSecondaryServices(backends=multi_backend_connection, processing=processing, config=config)
 
         service_types = implementation.service_types()
         assert service_types == single_service_type
 
     def test_service_types_simple_cached(
-        self, multi_backend_connection, config, catalog, backend1, backend2, requests_mock
+        self,
+        multi_backend_connection,
+        config,
+        catalog,
+        backend1,
+        backend2,
+        requests_mock,
     ):
         """Scenario: The service_types call is cached:
             When we get the service types several times, the second call that happens before the cache expires,
             doesn't hit the backend.
             But the third call that happens that happens after the cache has expired does hit the backend again.
         """
+        # Aggregator checks if the backend supports GET /service_types, so we have to mock that up too.
+        requests_mock.get(
+            backend1 + "/", json=JSON_CAPABILITIES_WITH_SERVICE_TYPES_SUPPORTED
+        )
+        requests_mock.get(
+            backend2 + "/", json=JSON_CAPABILITIES_WITH_SERVICE_TYPES_SUPPORTED
```

Comment:
shouldn't this be JSON_CAPABILITIES_NO_ENDPOINTS  ? because `/service_types`  is not mocked below

Reply:
You are right.  In fact, we only need backend1, so I have just removed that mock for backend2.